zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -435,15 +463,42 @@ impl ZebraDb {
         }
     }
 
+    /// Get the orchard note commitment subtress for the finalized tip.
+    #[allow(clippy::unwrap_in_result)]
+    fn orchard_subtree_for_tip(&self) -> Option<NoteCommitmentSubtree<orchard::tree::Node>> {
+        let tip_height = self.finalized_tip_height().unwrap_or(Height(0));
+        let orchard_activation_height = NetworkUpgrade::Nu5
+            .activation_height(self.network())
+            .expect("sapling activation height must exist");
+
+        if tip_height >= orchard_activation_height {
+            let orchard_nct_handle = self.db.cf_handle("orchard_note_commitment_tree").unwrap();
+
+            let tip_note_commitment_tree: Arc<orchard::tree::NoteCommitmentTree> = self
+                .db
+                .zs_last_key_value(&orchard_nct_handle)
+                .map(|(_key, value): (RawBytes, _)| Arc::new(value))
+                .expect("Orchard note commitment tree must exist if there is a finalized tip");
+
+            if let Some((subtree_index, node)) =
+                tip_note_commitment_tree.completed_subtree_index_and_root()
+            {
+                return Some(NoteCommitmentSubtree::new(subtree_index, tip_height, node));
+            }
+        }
+
+        None
```

Comment:
```suggestion
        let orchard_subtrees = self
            .db
            .cf_handle("orchard_note_commitment_subtree")
            .unwrap();

        let (index, subtree_data): (
            NoteCommitmentSubtreeIndex,
            NoteCommitmentSubtreeData<orchard::tree::Node>,
        ) = self.db.zs_last_key_value(&orchard_subtrees )?;

        Some(subtree_data.with_index(index))
```

Reply:
Let's hold off on Orchard changes until we've got the Sapling changes worked out?
https://github.com/ZcashFoundation/zebra/pull/7636/files#r1340708466